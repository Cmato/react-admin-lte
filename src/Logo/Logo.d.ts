import * as React from 'react';

export interface LogoProps {
  component?: React.ReactType<LogoProps>;
  large?: React.ReactNode;
  mini?: React.ReactNode;
}

declare const Logo: React.ComponentType<LogoProps>;

export default Logo;
