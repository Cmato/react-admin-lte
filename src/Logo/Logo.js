import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Logo = ({
  className: classNameProp,
  component: Component,
  large,
  mini,
  ...other
}) => {
  const className = classNames({
    logo: true
  }, classNameProp);

  return (
    <Component className={className} {...other}>
      <span className="logo-mini">{mini}</span>
      <span className="logo-lg">{large}</span>
    </Component>
  );
};

Logo.propTypes = {
  /**
   * @ignore
   */
  className: PropTypes.string,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  mini: PropTypes.node,
  large: PropTypes.node
};

Logo.defaultProps = {
  component: 'a'
};

export default Logo;
