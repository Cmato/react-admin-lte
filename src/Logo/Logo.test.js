import React from 'react';
import { shallow } from 'enzyme';
import Logo from './Logo';

it('renders default', () => {
  expect(shallow(<Logo />))
    .toMatchSnapshot();
});

it('renders mini logo', () => {
  expect(shallow(<Logo mini={<div>mini</div>} />))
    .toMatchSnapshot();
});

it('renders large logo', () => {
  expect(shallow(<Logo large={<div>large</div>} />))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<Logo component="Link" />))
    .toMatchSnapshot();
});


