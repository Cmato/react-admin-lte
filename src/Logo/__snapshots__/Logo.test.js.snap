// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom component 1`] = `
<Link
  className="logo"
>
  <span
    className="logo-mini"
  />
  <span
    className="logo-lg"
  />
</Link>
`;

exports[`renders default 1`] = `
<a
  className="logo"
>
  <span
    className="logo-mini"
  />
  <span
    className="logo-lg"
  />
</a>
`;

exports[`renders large logo 1`] = `
<a
  className="logo"
>
  <span
    className="logo-mini"
  />
  <span
    className="logo-lg"
  >
    <div>
      large
    </div>
  </span>
</a>
`;

exports[`renders mini logo 1`] = `
<a
  className="logo"
>
  <span
    className="logo-mini"
  >
    <div>
      mini
    </div>
  </span>
  <span
    className="logo-lg"
  />
</a>
`;
